OOM but test passes. Following up to fix.

```
[2025-07-14T16:52:34Z] INFO 07-14 09:52:34 [model_runner.py:1171] Starting to load model deepseek-ai/DeepSeek-V2-Lite-Chat...
[2025-07-14T16:52:35Z] DEBUG 07-14 09:52:35 [decorators.py:110] Inferred dynamic dimensions for forward method of <class 'vllm.model_executor.models.deepseek_v2.DeepseekV2Model'>: ['input_ids', 'positions', 'intermediate_tensors', 'inputs_embeds']
[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458] CUDA out of memory. Tried to allocate 704.00 MiB. GPU 0 has a total capacity of 22.05 GiB of which 508.12 MiB is free. Including non-PyTorch memory, this process has 21.54 GiB memory in use. Of the allocated memory 21.30 GiB is allocated by PyTorch, and 28.75 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458] Traceback (most recent call last):

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 446, in run_mp_engine

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     engine = MQLLMEngine.from_vllm_config(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 133, in from_vllm_config

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     return cls(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]            ^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 87, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.engine = LLMEngine(*args, **kwargs)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/llm_engine.py", line 265, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.model_executor = executor_class(vllm_config=vllm_config)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/executor_base.py", line 287, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     super().__init__(*args, **kwargs)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/executor_base.py", line 53, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self._init_executor()

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/mp_distributed_executor.py", line 126, in _init_executor

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self._run_workers("load_model",

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/mp_distributed_executor.py", line 186, in _run_workers

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     driver_worker_output = run_method(self.driver_worker, sent_method,

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/utils/__init__.py", line 2943, in run_method

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     return func(*args, **kwargs)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]            ^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/worker.py", line 210, in load_model

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.model_runner.load_model()

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/model_runner.py", line 1174, in load_model

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.model = get_model(vllm_config=self.vllm_config)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/__init__.py", line 59, in get_model

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     return loader.load_model(vllm_config=vllm_config,

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/base_loader.py", line 38, in load_model

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     model = initialize_model(vllm_config=vllm_config,

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/utils.py", line 64, in initialize_model

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     return model_class(vllm_config=vllm_config, prefix=prefix)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 723, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.model = DeepseekV2Model(vllm_config=vllm_config,

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/decorators.py", line 152, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     old_init(self, vllm_config=vllm_config, prefix=prefix, **kwargs)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 661, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.start_layer, self.end_layer, self.layers = make_layers(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                                                     ^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/utils.py", line 640, in make_layers

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     maybe_offload_to_cpu(layer_fn(prefix=f"{prefix}.{idx}"))

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 663, in <lambda>

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     lambda prefix: DeepseekV2DecoderLayer(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                    ^^^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 570, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.mlp = DeepseekV2MoE(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                ^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 148, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.experts = FusedMoE(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                    ^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/layers/fused_moe/layer.py", line 772, in __init__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     self.quant_method.create_weights(layer=self, **moe_quant_params)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/layers/fused_moe/layer.py", line 263, in create_weights

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     w13_weight = torch.nn.Parameter(torch.empty(

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]                                     ^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_device.py", line 104, in __torch_function__

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]     return func(*args, **kwargs)

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458]            ^^^^^^^^^^^^^^^^^^^^^

[2025-07-14T16:52:35Z] ERROR 07-14 09:52:35 [engine.py:458] torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 704.00 MiB. GPU 0 has a total capacity of 22.05 GiB of which 508.12 MiB is free. Including non-PyTorch memory, this process has 21.54 GiB memory in use. Of the allocated memory 21.30 GiB is allocated by PyTorch, and 28.75 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
[2025-07-14T16:52:35Z] Process SpawnProcess-1:
[2025-07-14T16:52:35Z] Traceback (most recent call last):
[2025-07-14T16:52:35Z]   File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
[2025-07-14T16:52:35Z]     self.run()
[2025-07-14T16:52:35Z]   File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
[2025-07-14T16:52:35Z]     self._target(*self._args, **self._kwargs)
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 460, in run_mp_engine
[2025-07-14T16:52:35Z]     raise e from None
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 446, in run_mp_engine
[2025-07-14T16:52:35Z]     engine = MQLLMEngine.from_vllm_config(
[2025-07-14T16:52:35Z]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 133, in from_vllm_config
[2025-07-14T16:52:35Z]     return cls(
[2025-07-14T16:52:35Z]            ^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 87, in __init__
[2025-07-14T16:52:35Z]     self.engine = LLMEngine(*args, **kwargs)
[2025-07-14T16:52:35Z]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/llm_engine.py", line 265, in __init__
[2025-07-14T16:52:35Z]     self.model_executor = executor_class(vllm_config=vllm_config)
[2025-07-14T16:52:35Z]                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/executor_base.py", line 287, in __init__
[2025-07-14T16:52:35Z]     super().__init__(*args, **kwargs)
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/executor_base.py", line 53, in __init__
[2025-07-14T16:52:35Z]     self._init_executor()
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/mp_distributed_executor.py", line 126, in _init_executor
[2025-07-14T16:52:35Z]     self._run_workers("load_model",
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/mp_distributed_executor.py", line 186, in _run_workers
[2025-07-14T16:52:35Z]     driver_worker_output = run_method(self.driver_worker, sent_method,
[2025-07-14T16:52:35Z]                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/utils/__init__.py", line 2943, in run_method
[2025-07-14T16:52:35Z]     return func(*args, **kwargs)
[2025-07-14T16:52:35Z]            ^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/worker.py", line 210, in load_model
[2025-07-14T16:52:35Z]     self.model_runner.load_model()
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/model_runner.py", line 1174, in load_model
[2025-07-14T16:52:35Z]     self.model = get_model(vllm_config=self.vllm_config)
[2025-07-14T16:52:35Z]                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/__init__.py", line 59, in get_model
[2025-07-14T16:52:35Z]     return loader.load_model(vllm_config=vllm_config,
[2025-07-14T16:52:35Z]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/base_loader.py", line 38, in load_model
[2025-07-14T16:52:35Z]     model = initialize_model(vllm_config=vllm_config,
[2025-07-14T16:52:35Z]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/model_loader/utils.py", line 64, in initialize_model
[2025-07-14T16:52:35Z]     return model_class(vllm_config=vllm_config, prefix=prefix)
[2025-07-14T16:52:35Z]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 723, in __init__
[2025-07-14T16:52:35Z]     self.model = DeepseekV2Model(vllm_config=vllm_config,
[2025-07-14T16:52:35Z]                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/decorators.py", line 152, in __init__
[2025-07-14T16:52:35Z]     old_init(self, vllm_config=vllm_config, prefix=prefix, **kwargs)
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 661, in __init__
[2025-07-14T16:52:35Z]     self.start_layer, self.end_layer, self.layers = make_layers(
[2025-07-14T16:52:35Z]                                                     ^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/utils.py", line 640, in make_layers
[2025-07-14T16:52:35Z]     maybe_offload_to_cpu(layer_fn(prefix=f"{prefix}.{idx}"))
[2025-07-14T16:52:35Z]                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 663, in <lambda>
[2025-07-14T16:52:35Z]     lambda prefix: DeepseekV2DecoderLayer(
[2025-07-14T16:52:35Z]                    ^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 570, in __init__
[2025-07-14T16:52:35Z]     self.mlp = DeepseekV2MoE(
[2025-07-14T16:52:35Z]                ^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v2.py", line 148, in __init__
[2025-07-14T16:52:35Z]     self.experts = FusedMoE(
[2025-07-14T16:52:35Z]                    ^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/layers/fused_moe/layer.py", line 772, in __init__
[2025-07-14T16:52:35Z]     self.quant_method.create_weights(layer=self, **moe_quant_params)
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/layers/fused_moe/layer.py", line 263, in create_weights
[2025-07-14T16:52:35Z]     w13_weight = torch.nn.Parameter(torch.empty(
[2025-07-14T16:52:35Z]                                     ^^^^^^^^^^^^
[2025-07-14T16:52:35Z]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_device.py", line 104, in __torch_function__
[2025-07-14T16:52:35Z]     return func(*args, **kwargs)
[2025-07-14T16:52:35Z]            ^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:35Z] torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 704.00 MiB. GPU 0 has a total capacity of 22.05 GiB of which 508.12 MiB is free. Including non-PyTorch memory, this process has 21.54 GiB memory in use. Of the allocated memory 21.30 GiB is allocated by PyTorch, and 28.75 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
[2025-07-14T16:52:36Z] [rank0]:[W714 09:52:36.281349739 ProcessGroupNCCL.cpp:1476] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
[2025-07-14T16:52:37Z] DEBUG 07-14 09:52:37 [client.py:261] Shutting down MQLLMEngineClient output handler.
[2025-07-14T16:52:37Z] Traceback (most recent call last):
[2025-07-14T16:52:37Z]   File "/usr/local/bin/vllm", line 10, in <module>
[2025-07-14T16:52:37Z]     sys.exit(main())
[2025-07-14T16:52:37Z]              ^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/cli/main.py", line 65, in main
[2025-07-14T16:52:37Z]     args.dispatch_function(args)
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/cli/serve.py", line 57, in cmd
[2025-07-14T16:52:37Z]     uvloop.run(run_server(args))
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/uvloop/__init__.py", line 109, in run
[2025-07-14T16:52:37Z]     return __asyncio.run(
[2025-07-14T16:52:37Z]            ^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/lib/python3.12/asyncio/runners.py", line 195, in run
[2025-07-14T16:52:37Z]     return runner.run(main)
[2025-07-14T16:52:37Z]            ^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/lib/python3.12/asyncio/runners.py", line 118, in run
[2025-07-14T16:52:37Z]     return self._loop.run_until_complete(task)
[2025-07-14T16:52:37Z]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "uvloop/loop.pyx", line 1518, in uvloop.loop.Loop.run_until_complete
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/uvloop/__init__.py", line 61, in wrapper
[2025-07-14T16:52:37Z]     return await main
[2025-07-14T16:52:37Z]            ^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/openai/api_server.py", line 1675, in run_server
[2025-07-14T16:52:37Z]     await run_server_worker(listen_address, sock, args, **uvicorn_kwargs)
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/openai/api_server.py", line 1695, in run_server_worker
[2025-07-14T16:52:37Z]     async with build_async_engine_client(args, client_config) as engine_client:
[2025-07-14T16:52:37Z]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/lib/python3.12/contextlib.py", line 210, in __aenter__
[2025-07-14T16:52:37Z]     return await anext(self.gen)
[2025-07-14T16:52:37Z]            ^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/openai/api_server.py", line 158, in build_async_engine_client
[2025-07-14T16:52:37Z]     async with build_async_engine_client_from_engine_args(
[2025-07-14T16:52:37Z]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/lib/python3.12/contextlib.py", line 210, in __aenter__
[2025-07-14T16:52:37Z]     return await anext(self.gen)
[2025-07-14T16:52:37Z]            ^^^^^^^^^^^^^^^^^^^^^
[2025-07-14T16:52:37Z]   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/openai/api_server.py", line 291, in build_async_engine_client_from_engine_args
[2025-07-14T16:52:37Z]     raise RuntimeError(
[2025-07-14T16:52:37Z] RuntimeError: Engine process failed to start. See stack trace for the root cause.
[2025-07-14T16:52:39Z] [32mPASSED[0m
```

--- a/tests/distributed/test_pipeline_parallel.py
+++ b/tests/distributed/test_pipeline_parallel.py
@@ -339,6 +339,7 @@ def _compare_tp(
         common_args.extend(["--hf-overrides", json.dumps(hf_overrides)])
 
     specific_case = tp_size == 2 and pp_size == 2 and chunked_prefill
+    testing_ray_compiled_graph = False
     if distributed_backend == "ray" and (vllm_major_version == "1"
                                          or specific_case):
         # For V1, test Ray Compiled Graph for all the tests

--- a/tests/distributed/test_pipeline_parallel.py
+++ b/tests/distributed/test_pipeline_parallel.py
@@ -14,8 +14,9 @@ from typing import Literal, NamedTuple, Optional
 
 import pytest
 
-from vllm.config import TaskOption
+from vllm.config import _FLOAT16_NOT_SUPPORTED_MODELS, TaskOption
 from vllm.logger import init_logger
+from vllm.transformers_utils.config import get_config
 
 from ..models.registry import HF_EXAMPLE_MODELS
 from ..utils import compare_two_settings, create_new_process_for_each_test
@@ -288,6 +289,11 @@ def _compare_tp(
     trust_remote_code = model_info.trust_remote_code
     tokenizer_mode = model_info.tokenizer_mode
     hf_overrides = model_info.hf_overrides
+    hf_config = get_config(model_id, trust_remote_code)
+
+    dtype = "float16"
+    if hf_config.model_type in _FLOAT16_NOT_SUPPORTED_MODELS:
+        dtype = "bfloat16"
 
     if load_format == "dummy":
         # Avoid OOM
@@ -317,7 +323,7 @@ def _compare_tp(
     common_args = [
         # use half precision for speed and memory savings in CI environment
         "--dtype",
-        "float16",
+        dtype,
         "--max-model-len",
         "2048",
         "--max-num-seqs",

--- a/tests/distributed/test_pipeline_parallel.py
+++ b/tests/distributed/test_pipeline_parallel.py
@@ -248,6 +248,7 @@ TEST_MODELS = [
     "meta-llama/Llama-3.2-1B-Instruct",
     "ArthurZ/Ilama-3.2-1B",
     "ibm/PowerLM-3b",
+    "deepseek-ai/DeepSeek-V2-Lite-Chat",
     # [LANGUAGE EMBEDDING]
     "intfloat/e5-mistral-7b-instruct",
     "BAAI/bge-multilingual-gemma2",

